test/envtest/setup.go

```diff
@@ -80,5 +80,9 @@ func Setup(t *testing.T, scheme *runtime.Scheme) *rest.Config {
 		wg.Wait()
 	})
 
+	// Set the env vars that the controller expects (not using t.Setenv because envtests are run in parallel).
+	_ = os.Setenv("POD_NAMESPACE", "default")            //nolint:tenv
+	_ = os.Setenv("POD_NAME", "kong-ingress-controller") //nolint:tenv
```

Comment:
🤔 I'm not sure about this. This way all the tests that are run will receive this and it will never be cleaned. This might potentially be problematic.

Do we need this or is it a nice to have to prevent warning logs?

Reply:
Yeah, I wanted to get rid of the noisy logs. How about doing it this way: https://github.com/Kong/kubernetes-ingress-controller/pull/4277 then?